### DISCUSSION TITLE
useExpande and useGlobalFilter strange behavior

### DISCUSSION THREAD

**Discussion by ZiedHf** [TanStack/table#3745] [Q&A]
An example in CodeSanbox below.

Whenever I use `React.useMemo` for the first options parameter of `useTable`, I get this strange behavior in my component.

I am trying to filter the Parent Rows and theirs Leafs as well.

But when I filter a leaf name, then I delete the filter input value, it looks like that `filterRows` in `useGlobalFilter` uses an old instance of rows with filtered Leafs.

I am confused, is there something I am doing wrong ?

It's work as expected without `React.useMemo` but I am trying to enhance the performance of my component since it became a little bit slow.

[Exmaple here](https://codesandbox.io/s/react-table-expanded-global-filter-ehm-r2uxrx)

**Comment by sumitaney** [comment]
@ZiedHf Try using useEffect for setColumns, instead of react useMemo
